vignettes/Surveys_Walkthrough.Rmd

```diff
@@ -661,6 +930,725 @@ study_data_cleaned <- study_data_cleaned %>%
+  set_variable_labels(
+    ## variable name = variable label 
+    age_years     = "Age (years)",
+    age_group     = "Age group (years)",
+    age_months    = "Age (months)", 
+    age_group_mon = "Age group (months)",
+    age_category  = "Age category", 
+    died          = "Died", 
+    startcause    = "Observation start", 
+    endcause      = "Observation end"
+    )
+```
+
+
+
+
+### Remove unused data  
+
+This chunk contains code to drop and examine rows for which there is no consent, there are missing start or end dates, village name (town A or B) is "Other", age is missing, or sex is missing. These rows are first saved as `dropped`. This allows you to inspect them before actually dropping the rows.
+
+In the second command, an `anti_join()` is used to actually drop the rows from `study_data_cleaned`. An anti join is a "filtering join" which looks in the `dropped` data frame and leaves only rows in `study_data_cleaned` which are *not* present in `dropped`. Its a fancy way of doing a filter, but allowing you the opportunity to inspect the rows first.  
+
+
+```{r, eval=FALSE}
+## store the cases that you drop so you can describe them (e.g. non-consenting 
+## or wrong village/cluster)
+dropped <- study_data_cleaned %>% 
+  filter(!consent | is.na(startdate) | is.na(enddate) | 
+           ## note that whatever you use for weights cannot be missing!
+           is.na(village_name) | is.na(age_years) | is.na(sex))
+
+## use the dropped cases to remove the unused rows from the survey data set  
+study_data_cleaned <- anti_join(study_data_cleaned, dropped, by = names(dropped))
+```
+
+
+### Drop duplicates  
+
+Deduplication is a complicated topic. We suggest you review the Epi R Handbook chapter on [Deduplication](https://epirhandbook.com/en/de-duplication.html) if you want to seriously approach this topic.
+
+In this code chunk, a simple deduplication command is offered using the `distinct()` function. In this example, for any rows that have identical `uid`, `sex`, AND `age_group`, only the first row is kept. The `.keep_all = TRUE` refers to the *columns* and ensures that all columns are kept, not only the ones you used to assess duplicates.  
+
+
+```{r}
+## option 1: only keep the first occurrence of duplicate case 
+study_data_cleaned <- study_data_cleaned %>% 
+  ## find duplicates based on case number, sex and age group 
+  ## only keep the first occurrence 
+  distinct(uid, sex, age_group, .keep_all = TRUE)
+  
+```
+
+A second option available to you is commented out. This option allows you to inspect duplicates before they are filtered/removed.  
+
+
+### Survey weights
+
+These data come from a cluster survey, but because we have population counts for age and sex, we are able to make adjustments to better reflect the population, and therefore produce stratified estimates by age and sex. 
+
+At the end of the first command, we remove "health_district" because we only want to stratify by age and sex.  
```

Comment:
no - we decided that this was indeed a stratified cluster and we can include health_district!

Reply:
fixed